Feat/create prize

qnmgjeklnbrl opened the pull request:
- 예외처리 구조 변경했습니다.
- 이전 버전 서비스클래스 save()메서드가 responseDto를 통째로 반환했는데 데이터 쓸데없이 무겁게 쓰는것 같아서 id만 반환하게 수정했습니다.
- 입력값이 정상, 비정상일때 동작 테스트 했습니다.

qnmgjeklnbrl committed:
프로젝트 설정 및 mysql의존성 추가

qnmgjeklnbrl committed:
feat: auditing, 예외처리 구조 변경

qnmgjeklnbrl committed:
test: Prize 저장 테스트

qnmgjeklnbrl committed:
feat: Prize를 저장할 수 있다

dlehddn commented on prize/src/main/java/ddalkak/prize/service/impl/PrizeServiceImpl.java:
```diff
@@ -0,0 +1,41 @@
+        static Long ofRange(Long range) {
+            // 1 ~ range 범위 난수 생성
+            return (long) ( Math.random() * range )+ 1;
+        }
```
SecureRandom을 사용하는게 어떠실지요
Random보다 보안적으로 강화되고 중복 랜덤 수 발행이 적다고 합니다.
(Math.random()은 내부적으로 Random을 활용합니다.)

dlehddn commented on prize/src/test/java/ddalkak/prize/controller/PrizeControllerTest.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    @Test
+    @DisplayName("잘못된 입력값이면 InvalidInputException 발생")
```
ParameterizedTest을 활용해보세요
이처럼 여러 상황에서 실패를 검증해야하는 경우, 다양한 상황의 input을 넣고 하나의 테스트 코드에서 전부 수행할 수 있습니다.

dlehddn commented on prize/src/main/java/ddalkak/prize/domain/dto/PrizeRequestDto.java:
```diff
@@ -0,0 +1,31 @@
+        Long probabilityRange
+
+){
+        public Prize toEntity(Long randomNumber) {
```
음 RequestDto가 toEntity()의 책임을 지는게 적절한지 생각해봐야겠는데요,
저는 RequestDto는 그 자체로 입력값 파싱 및 검증이 끝이라고 생각합니다.
엔티티 자체에서 생성 책임을 갖는건 어떻게 생각하시나요?

dlehddn reviewed (commented):
고생하셨습니다. 코멘트 읽어보시고 답변 주세요:)

qnmgjeklnbrl committed:
chore: ParameterizedTest로 테스트 구조 변경

qnmgjeklnbrl committed:
refactor: 요청사항 반영

- Math.ramdom() -> secureRandom사용으로 변경
- 리포지토리 구조 변경
- requestDto -> prize엔티티로 파싱할때 prize생성자 사용으로 변경

qnmgjeklnbrl commented:
피드백 반영해서 다시 커밋 했습니다. 확인 부탁드립니다 ㅎㅎ

dlehddn commented on prize/src/main/java/ddalkak/prize/domain/dto/PrizeRequestDto.java:
```diff
@@ -9,7 +10,7 @@
 public record PrizeRequestDto (
 
-        @NotNull
+        @NotNull@NotBlank
```
NotBlank는 NotNull을 포함하는 개념입니다ㅎ
NotNull은 지우셔도 됩니다.

dlehddn commented on prize/src/main/java/ddalkak/prize/service/impl/PrizeServiceImpl.java:
```diff
@@ -2,26 +2,35 @@
 public class PrizeServiceImpl implements PrizeService {
     @Autowired
-    private final PrizeRepository prizeRepository;
+    private final PrizeJpaRepository prizeRepository;
```
위에 새롭게 PrizeRepositoryImpl 만드셨는데 까먹으신듯요
PrizeRepository 타입으로 받으시려고 한거죠??

qnmgjeklnbrl replied:
네 ㅎ

dlehddn reviewed (commented):
잘못 작성하신거 있는거 같아 코멘트 남깁니다ㅎ

qnmgjeklnbrl committed:
refactor: 요청사항 반영

qnmgjeklnbrl commented:
수정했습니다

dlehddn reviewed (approved):
고생하셨습니다~
